cocotb/handle.py

```diff
@@ -577,6 +577,31 @@ def loads(self):
         """An iterator for gathering all loads on a signal."""
         return _SimIterator(self._handle, simulator.LOADS)
 
+class _SetAction:
+    """Base class representing the type of action used while write-accessing a handle."""
+    pass
+
+class _SetValueAction(_SetAction):
+    __slots__ = ("value",)
+    """Base class representing the type of action used while write-accessing a handle with a value."""
+    def __init__(self, value):
+        self.value = value
+
+class Deposit(_SetValueAction):
+    """Action used for placing a value into a given handle."""
+    _action = 0
```

Comment:
As suggested below in https://github.com/cocotb/cocotb/pull/1403/files#r380357823:
```suggestion
    def _as_gpi_args_for(self, hdl):
        return self.value, 0
```